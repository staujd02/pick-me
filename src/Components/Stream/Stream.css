iframe {
  border: 0; 
  width: 100%; 
  height: -webkit-fill-available;
}

div {
  height: 100%;
}

::-webkit-scrollbar { 
  display: none; 
}

body{
  margin: 0;
}