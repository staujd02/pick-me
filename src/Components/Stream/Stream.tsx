import React from 'react';
import './Stream.css';

class Stream extends React.Component<StreamProps> {

    noOpMessage(){
        return "Your browser doesn't support iFrames.";
    }

    render(){
        const { source, title } = this.props.source;
        return (
            <iframe title={title} allow={"accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"} src={source}>{this.noOpMessage()}</iframe>
        );
    }

}

export default Stream;