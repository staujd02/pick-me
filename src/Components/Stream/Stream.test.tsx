import React from 'react';
import Stream from './Stream';
import { StreamState } from 'http2';
import { ShallowWrapper, shallow } from 'enzyme';
import { formatHTML } from '../../TestUtilities/htmlFormatter';

describe('The Stream', () => {

    const Source = {
        title: "Google",
        source: "https://www.google.com"
    } as Source;
    let wrapper: ShallowWrapper<StreamProps, StreamState>;

    beforeEach(() => {
        wrapper = shallow(<Stream source={Source} />)
    });

    it('renders correctly', () => expect(formatHTML(wrapper.html())).toMatchSnapshot());
});