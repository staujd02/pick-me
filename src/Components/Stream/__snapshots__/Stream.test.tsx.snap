// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Stream renders correctly 1`] = `
"
<iframe title=\\"Google\\" allow=\\"accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture\\" src=\\"https://www.google.com\\">Your browser doesn&#x27;t support iFrames.</iframe>
"
`;
